src/cpp/core/r_util/RProjectFile.cpp

```diff
@@ -311,6 +312,54 @@ std::ostream& operator << (std::ostream& stream, const YesNoAskValue& val)
    return stream ;
 }
 
+Error findProjectFile(FilePath filePath,
+                      FilePath* pProjPath)
+{
+   // check to see if we already have a .Rproj file
+   if (filePath.extension() == ".Rproj")
+   {
+      *pProjPath = filePath;
+      return Success();
+   }
+   
+   if (!filePath.isDirectory())
+      filePath = filePath.parent();
+   
+   // no .Rproj file found; scan parent directories
+   for (; filePath.exists(); filePath = filePath.parent())
```

Comment:
Wait, sorry -- the APIs here are actually different in how they respond to errors. The API I'm using actually lives here (the version with no arguments):

https://github.com/rstudio/rstudio/blob/master/src/cpp/core/FilePath.cpp#L316-L328

and that API explicitly does not log errors that occur due to missing permissions.